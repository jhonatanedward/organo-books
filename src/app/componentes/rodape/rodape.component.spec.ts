import { ComponentFixture, TestBed } from "@angular/core/testing"
import { RodapeComponent } from "./rodape.component"

describe('Rodape component', () => {
    let fixture: ComponentFixture<RodapeComponent>
    let component: RodapeComponent
    
    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                RodapeComponent
            ]
        })

        fixture = TestBed.createComponent(RodapeComponent)
        component = fixture.componentInstance
    })

    it('deveria ser criado', () => {
        expect(component).toBeTruthy()
    })

    it('deveria definir os atributos src e alt', () => {
        expect(component.src).toBeDefined()
        expect(component.alt).toBeDefined()
    })

    it('deveria renderizar o rodape de acordo com as propriedades src e alt', () => {
        component.src = 'https://example.com/test-image.jpg'
        component.alt = 'Imagem teste'
        expect(component).toMatchSnapshot()
    })
})