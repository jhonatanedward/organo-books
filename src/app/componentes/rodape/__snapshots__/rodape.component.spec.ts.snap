// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rodape component deveria renderizar o rodape de acordo com as propriedades src e alt 1`] = `
RodapeComponent {
  "__ngContext__": 4,
  "alt": "Imagem teste",
  "src": "https://example.com/test-image.jpg",
}
`;
